src/ConstructiveSolidGeometry/Intervals.jl

```diff
@@ -0,0 +1,8 @@
+@inline _left(r::T) where {T <: Real} = -r
+@inline _left(r::AbstractInterval) = r.left
+
+@inline _right(r::T) where {T <: Real} = r
+@inline _right(r::AbstractInterval) = r.right
+
+@inline _width(z::T) where {T <: Real} = T(2) * z
```

Comment:
I believe the compiler is clever enough that you can just right `2z`. 
Right @oschulz ?

Reply:
If `T` is an `Int32`, `2 * z` gives and `Int64` and `T(2) * z` gives an `Int32`.
For all `Float` it seems to make no difference.

Reply:
Yes, because `promote_type(Int32, Int64) = Int64` and `promote_type(IntX, FloatY) = FloatY`. 
`f(x) = 2x; @code_native f(2.3)` gives me almost the same as `g(x::T) where {T} = T(2) * x; @code_native g(2.3)` 
Also almost the same for `@code_llvm`. As far as I understand the output, the compiler handles it automatically and we can use `2z` here (and in general in such situations). But I am not entirely sure.